### PR TITLE
Add IP range Input support Issue #98

### PR DESCRIPTION
**Pull request by ShubhamRasal**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -183,6 +183,42 @@ In order to list CIDR blocks for given list of IPs, use the following command.
...
+192.168.0.16/28
+192.168.0.32/27
+192.168.0.64/26
+192.168.0.128/25
+
+```
+
+It is also possible to pass the list of IP ranges. The final output will contain the CIDR prefixes of given ranges.
+```
+$ mapcidr -cl 192.168.0.1-192.168.0.255,2c0f:fec9::-2c0f:fed7:ffff:ffff:ffff:ffff:ffff:ffff  -aggregate
```
Comment (jdk2588) [inline README.md]: Do we need to have a list of IP range?

File: cmd/mapcidr/main.go [hunk 2]
```diff
@@ -374,12 +387,34 @@ func process(wg *sync.WaitGroup, chancidr, outputchan chan string) {
...
-		}
-		for _, cidrIPV6 := range cCidrsIPV6 {
-			outputchan <- cidrIPV6.String()
+		// Find out the possible CIDR prefixes from given IP address range
+		if options.CIDRFromIPRange {
+			// check if ipRange has more than 2 values
+			if len(ipRange)%2 == 1 {
+				gologger.Fatal().Msgf("IP range can not have more than 2 values.")
+			}
+			for i := 0; i < len(ipRange)-1; i = i + 2 {
```
Comment (jdk2588) [inline cmd/mapcidr/main.go]: This loops is not required, if we need only single IP range?

Comment (ShubhamRasal) [reply cmd/mapcidr/main.go]: Removed the for loop

File: README.md (hunk 1 above)
Comment (ShubhamRasal) [reply README.md]: Removed this block from the readme

File: cmd/mapcidr/main.go [hunk 3]
```diff
@@ -393,19 +393,18 @@ func process(wg *sync.WaitGroup, chancidr, outputchan chan string) {
...
-				}
-				cidrs := mapcidr.RangeToCIDRs(ipRange[i], ipRange[i+1])
-				sort.Slice(cidrs, func(i, j int) bool {
-					return bytes.Compare(cidrs[i].IP, cidrs[j].IP) < 0
-				})
-				for _, cidr := range cidrs {
-					outputchan <- cidr.String()
-				}
+			// check if range is valid or not
+			if bytes.Compare(ipRange[0], ipRange[1]) > 0 {
```
Comment (jdk2588) [inline cmd/mapcidr/main.go]: Would not the internal function take care of this? Wondering :thinking: , if not then this logic should be in the function.

File: cmd/mapcidr/main.go [hunk 4]
```diff
@@ -374,12 +387,33 @@ func process(wg *sync.WaitGroup, chancidr, outputchan chan string) {
...
+				gologger.Fatal().Msgf("Start IP:%s must be less than End IP:%s", ipRange[0], ipRange[1])
+			}
+			cidrs := mapcidr.RangeToCIDRs(ipRange[0], ipRange[1])
+			sort.Slice(cidrs, func(i, j int) bool {
+				return bytes.Compare(cidrs[i].IP, cidrs[j].IP) < 0
+			})
+			for _, cidr := range cidrs {
+				outputchan <- cidr.String()
+			}
+			options.CIDRFromIPRange = false
```
Comment (jdk2588) [inline cmd/mapcidr/main.go]: What is the purpose of setting this to false?

Comment (ShubhamRasal) [reply cmd/mapcidr/main.go]: I forgot to remove it. It was just for testing.

File: cmd/mapcidr/main.go (hunk 3 above)
Comment (ShubhamRasal) [reply cmd/mapcidr/main.go]: The internal function doesn't handle any errors. The input validation is handled in main.go.  All the fatal errors are main.go only.

Comment (ShubhamRasal) [reply cmd/mapcidr/main.go]: Updated the code and moved the IP comparing and sorting logic to internal function

**Review by jdk2588 — APPROVED** [review]

**Pull request merged**